plugins/discourse-local-dates/assets/javascripts/initializers/discourse-local-dates.js.es6

```diff
@@ -73,6 +73,33 @@ function _rangeElements(element) {
 
 function initializeDiscourseLocalDates(api) {
   const siteSettings = api.container.lookup("site-settings:main");
+  const chat = api.container.lookup("service:chat");
+
+  if (chat) {
+    chat.addToolbarButton({
+      title: "discourse_local_dates.title",
+      id: "local-dates",
+      icon: "calendar-alt",
+      action: "insertDiscourseLocalDate",
+    });
+
+    api.modifyClass("component:chat-composer", {
+      pluginId: "discourse-local-dates",
+      actions: {
+        insertDiscourseLocalDate() {
+          const chatComposer = this;
+          showModal("discourse-local-dates-create-modal").setProperties({
+            insertDate: (text) => {
+              const selected = chatComposer._getSelected(null, {
```

Comment:
It looks like you are calling a private API here. Is there a way you could make a public API for this?

Reply:
@eviltrout this is true, but this is modifying the component that those are private to... so technically this should be allowed, right?

Reply:
Unfortunately not. No plugin should be calling a method with `_` -- if I see a method with `_` I don't even search our `all-the-plugins` before changing it. You will need to establish a more public interface for what you are doing.